runner/runners/bazel.go

```diff
@@ -28,24 +28,45 @@ import (
 	bzsnapshot "github.com/twitter/scoot/snapshot/bazel"
 )
 
-// Preliminary setup for handling Bazel commands - verify Filer and populate command
-func preProcessBazel(filer snapshot.Filer, cmd *runner.Command) error {
+// Preliminary setup for handling Bazel commands - verify Filer and populate command.
+// Can return a bazelapi.ActionResult if a result was found already in the ActionCache
+func preProcessBazel(filer snapshot.Filer, cmd *runner.Command) (*bazelapi.ActionResult, error) {
 	bzFiler, ok := filer.(*bzsnapshot.BzFiler)
 	if !ok {
-		return fmt.Errorf("Filer could not be asserted as type BzFiler. Type is: %s", reflect.TypeOf(filer))
+		return nil, fmt.Errorf("Filer could not be asserted as type BzFiler. Type is: %s", reflect.TypeOf(filer))
 	}
 	if cmd.ExecuteRequest == nil {
-		return fmt.Errorf("Nil ExecuteRequest data in Command with RunType Bazel")
+		return nil, fmt.Errorf("Nil ExecuteRequest data in Command with RunType Bazel")
+	}
+
+	// Check for existing result in ActionCache
+	if !cmd.ExecuteRequest.GetRequest().GetSkipCacheLookup() {
+		log.Info("Checking for existing results for command in ActionCache")
+		ar, err := cas.GetCacheResult(bzFiler.CASResolver, cmd.ExecuteRequest.GetActionDigest())
+		if err != nil {
+			// Only treat as an error if we didn't get NotFoundError
+			if _, ok := err.(*cas.NotFoundError); !ok {
+				log.Errorf("Failed to check for cached result, will execute: %s", err)
+			}
```

Comment:
should there be a `return nil, err` here? (for the case that `err` isn't a `*cas.NotFoundError`

Reply:
Nah - no reason to fail the run - we just want to log that we weren't able to check the cache (metrics to come)

Reply:
whoops, missed the ! in !ok